SmartDeviceLinkTests/ProtocolSpecs/SDLProtocolReceivedMessageProcessorSpec.m

```diff
@@ -0,0 +1,554 @@
+                    Byte testByte = 0x10;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
+                });
+            });
+            context(@"with a frameType of SDLFrameTypeSingle", ^{
+                it(@"transitions to SERVICE_TYPE_STATE", ^{
+                    Byte testByte = 0x11;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
+                });
+            });
+            context(@"with a frameType of SDLFrameTypeFirst", ^{
+                it(@"transitions to SERVICE_TYPE_STATE", ^{
+                    Byte testByte = 0x12;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
+                });
+            });
+            context(@"with a frameType of SDLFrameTypeConsecutive", ^{
+                it(@"transitions to SERVICE_TYPE_STATE", ^{
+                    Byte testByte = 0x13;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
+                });
+            });
+            context(@"with an invalid frameType of 6", ^{
+                it(@"resets state to START_STATE", ^{
+                    Byte testByte = 0x66;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(START_STATE));
+                });
+            });
+        });
+    });
+    context(@"in SERVICE_TYPE_STATE", ^{
+        beforeEach(^{
+            testProcessor.state = SERVICE_TYPE_STATE;
+        });
+        context(@"When it recieves a SDLServiceTypeControl byte", ^{
```

Comment:
```suggestion
        context(@"when it receives a SDLServiceTypeControl byte", ^{
```

Always use lower-case, and please make sure the spelling of "receive" is correct throughout the file

Reply:
See above about context restructure, something like "when it receives a good service type byte"